### PR TITLE
Fix some erros in setup.sh

### PR DESCRIPTION
**Pull request by backwind1233**
## Changes
1. Give DISAMBIG_PREFIX a default value.
2. To prevent users from accidentally entering empty spaces.
3. Fix error while users configure default resource group and the the resource group don't exist in Azure.
4. Fix WLS_AKS_REPO_REF
  - Old value of WLS_AKS_REPO_REF: 364b7648b
  - New value of WLS_AKS_REPO_REF: 042a47ae4
  The way to get replicas in setupDBConnections.sh between commit [364b7648b](https://github.com/azure-javaee/weblogic-azure/blob/364b7648bbe395cb17683180401d07a3029abe91/weblogic-azure-aks/src/main/arm/scripts/setupDBConnections.sh#L195-L196) and commit [042a47ae4](https://github.com/azure-javaee/weblogic-azure/blob/042a47ae4dad755dfafdb30cfbdb753a24194984/weblogic-azure-aks/src/main/arm/scripts/setupDBConnections.sh#L202-L203)  are different, commit[364b7648b](https://github.com/azure-javaee/weblogic-azure/blob/364b7648bbe395cb17683180401d07a3029abe91/weblogic-azure-aks/src/main/arm/scripts/setupDBConnections.sh#L195-L196) not work.

## Test
✅ Github actions [passed](https://github.com/backwind1233/cargotracker-wls-aks/actions/runs/5873649759). 
image

### CODE REVIEW

File: .github/workflows/setup.sh [hunk 1]
```diff
@@ -55,21 +55,26 @@ msg() {
 
 setup_colors
 
-read -r -p "Enter a disambiguation prefix (try initials with a sequence number, such as ejb01): " DISAMBIG_PREFIX
+read -r -p "Enter a disambiguation prefix (try initials with a sequence number, default is ejb01): " DISAMBIG_PREFIX
```
Comment (backwind1233) [inline .github/workflows/setup.sh:58]: 1. Give DISAMBIG_PREFIX a default value.

File: .github/workflows/setup.sh [hunk 2]
```diff
@@ -55,21 +55,26 @@ msg() {
...
-  msg "${RED}You must enter a disambiguation prefix."
-  exit 1;
+if [ -z "$DISAMBIG_PREFIX" ]; then
+  DISAMBIG_PREFIX="ejb01"
+  echo "Using disambiguation prefix: $DISAMBIG_PREFIX"
 fi
 
+
 # get ORC_SSOUSER if not set at the beginning of this file
-if [ "$ORC_SSOUSER" == '' ] ; then
```
Comment (backwind1233) [inline .github/workflows/setup.sh:66]: To prevent users from accidentally entering empty spaces.

File: .github/workflows/setup.sh [hunk 3]
```diff
@@ -55,21 +55,26 @@ msg() {
...
+if [[ -z "$ORC_SSOUSER" ]]; then
+    while [[ -z "$ORC_SSOUSER" ]]; do
+        read -r -p "Enter Oracle single sign-on userid: " ORC_SSOUSER
+    done
 fi
 
 # get ORC_SSOPSW if not set at the beginning of this file
-if [ "$ORC_SSOPSW" == '' ] ; then
-    read -s -r -p "Enter password for preceding Oracle single sign-on userid: " ORC_SSOPSW
+if [[ -z "$ORC_SSOPSW" ]]; then
```
Comment (backwind1233) [inline .github/workflows/setup.sh:74]: To prevent users from accidentally entering empty spaces.

File: .github/workflows/setup.sh [hunk 4]
```diff
@@ -146,7 +151,7 @@ SP_OBJECT_ID_ARRAY=$(az ad sp list --display-name ${SERVICE_PRINCIPAL_NAME} --qu
 SP_OBJECT_ID_ARRAY=$(echo ${SP_OBJECT_ID_ARRAY} | xargs) || true
 SP_OBJECT_ID_ARRAY=${SP_OBJECT_ID_ARRAY//[/}
 SP_OBJECT_ID=${SP_OBJECT_ID_ARRAY//]/}
-az role assignment create --assignee ${SP_OBJECT_ID} --role "User Access Administrator" --subscription "${SUBSCRIPTION_ID}"
+az role assignment create --assignee ${SP_OBJECT_ID} --role "User Access Administrator" --subscription "${SUBSCRIPTION_ID}"  --scope "/subscriptions/${SUBSCRIPTION_ID}"
```
Comment (backwind1233) [inline .github/workflows/setup.sh:154]: Fix error while users configure default resource group and the the resource group don't exist in Azure.

**Review by edburns — APPROVED** [review]

**Pull request merged**